js/src/lib/components/InferenceWidget/shared/WidgetWrapper/WidgetWrapper.svelte

```diff
@@ -64,6 +66,21 @@
 		(async () => {
 			modelLoadInfo = await getModelLoadInfo(apiUrl, model.id, includeCredentials);
 			$modelLoadStates[model.id] = modelLoadInfo;
+
+			// run widget example
+			if (exampleQueryParams.length) {
+				const example = {} as TWidgetExample;
+				for (const key of exampleQueryParams) {
+					const val = getQueryParamVal(key);
+					example[key] = val;
+				}
+				applyInputSample(example);
+			} else {
+				const example = getWidgetExample<TWidgetExample>(model, validateExample);
+				if (callApiOnMount && example) {
+					applyInputSample(example, { inferenceOpts: { isOnLoadCall: true } });
+				}
+			}
```

Comment:
If I'm not mistaken, `getWidgetExample` here is never called when `exampleQueryParams` is set in the concrete Widget component (ie most widgets)

Reply:
Ok I tested it and I can confirm this behavior!

Reply:
Is there anything that needs to be changed here?

Reply:
> Is there anything that needs to be changed here?

yes

> I think this is the existing behaviour on main

No

Sorry I think my wording was not clear here. I'll try to reformulate.


## On main (hf.co)

When accessing for example HuggingFaceH4/zephyr-7b-alpha, The widget picksan example at random and starts generating (expected behavior).

![image](https://github.com/huggingface/hub-docs/assets/33657802/a468f623-c728-41e0-bce8-dcfa540a0b0f)


## On this branch

The widget does not try to get the example from the `widgetData` and displays this error (bug?).
This is because `exampleQueryParams.length` is always truthy in the TextGenerationWidget, so the `getWidgetExample` call get skipped

![image](https://github.com/huggingface/hub-docs/assets/33657802/5fdba943-5c1d-45fb-b034-57cf6a62c581)

------
Let me know if anything remains unclear 🤗

Reply:
indeed, fixed the logic https://github.com/huggingface/hub-docs/pull/1023/commits/7ebca52223d4beb9f15fc61d11c0dde5ff5fa362